hudi-spark-datasource/hudi-spark3-extensions_2.12/src/main/scala/org/apache/spark/sql/catalyst/analysis/HudiOperationsCheck.scala

```diff
@@ -0,0 +1,125 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import org.apache.hudi.execution.HudiSQLUtils
+import org.apache.spark.sql.{AnalysisException, SparkSession}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.expressions.{Expression, InSubquery, Literal, Not}
+import org.apache.spark.sql.catalyst.merge.HudiMergeIntoTable
+import org.apache.spark.sql.catalyst.plans.logical.{DeleteFromTable, LogicalPlan}
+import org.apache.spark.sql.execution.command._
+
+/**
+  * check rule which can intercept unSupportOperation for hudi
+  */
+class HudiOperationsCheck(spark: SparkSession) extends (LogicalPlan => Unit) {
+  val catalog = spark.sessionState.catalog
+
+  override def apply(plan: LogicalPlan): Unit = {
+    plan foreach {
+      case AlterTableAddPartitionCommand(tableName, _, _) if (checkHudiTable(tableName)) =>
+        throw new AnalysisException("AlterTableAddPartitionCommand are not currently supported in HUDI")
+
+      case a: AlterTableDropPartitionCommand if (checkHudiTable(a.tableName)) =>
+        throw new AnalysisException("AlterTableDropPartitionCommand are not currently supported in HUDI")
+
+      case AlterTableChangeColumnCommand(tableName, _, _) if (checkHudiTable(tableName)) =>
+        throw new AnalysisException("AlterTableChangeColumnCommand are not currently supported in HUDI")
+
+      case AlterTableRenamePartitionCommand(tableName, _, _) if (checkHudiTable(tableName)) =>
+        throw new AnalysisException("AlterTableRenamePartitionCommand are not currently supported in HUDI")
+
+      case AlterTableRecoverPartitionsCommand(tableName, _) if (checkHudiTable(tableName)) =>
+        throw new AnalysisException("AlterTableRecoverPartitionsCommand are not currently supported in HUDI")
+
+      case AlterTableSetLocationCommand(tableName, _, _) if (checkHudiTable(tableName)) =>
+        throw new AnalysisException("AlterTableSetLocationCommand are not currently supported in HUDI")
+
+      case DeleteFromTable(target, Some(condition)) if (hasNullAwarePredicateWithNot(condition) && checkHudiTable(target)) =>
+        throw new AnalysisException("Null-aware predicate sub-squeries a are not currently supported for DELETE")
+
+      case HudiMergeIntoTable(target, _, _, _, noMatchedClauses, _) =>
+        noMatchedClauses.map { m =>
+          m.resolvedActions.foreach { action =>
+            if (action.targetColNameParts.size > 1) {
+              throw new AnalysisException(s"cannot insert nested values: ${action.targetColNameParts.mkString("<") }")
```

Comment:
would you please describe why would not insert nested values?

Reply:
create table nestedTable (keyid int,a struct<number:int,score:float>) using hudi.   now a is  a nested filed.
think that:
"merge into nestedTable as t using sourceTable as s " +
          "on t.keyid = s.keyid when not matched then insert (t.keyid, **t.a.score**,) values (s.keyid, s.score)"

we cannot insert  t.keyid= s.keyid,  t.a.socre = s.score  to nestedTable directly;  since  we only set partial filed(**t.a.score**) for a nested filed **t.a**, the other partial filed(**t.a.number**) is unkown